From f1a6760b73207523e7c9e458d0207393a456acf2 Mon Sep 17 00:00:00 2001
From: "leo.ding" <leo.ding@rock-chips.com>
Date: Mon, 6 Nov 2017 16:11:52 +0800
Subject: [PATCH] video: rockchip: vpu: set grf register

3126 need set grf to change hardware, and set
 register should via regmap_write

Change-Id: I49b9baf9cdd442689183d59cdc594f81bcc6bbba
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_service.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index fdcb65489658..89f90b53d213 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -820,9 +820,8 @@ static void vpu_service_power_on(struct vpu_subdev_data *data,
 
 #define BIT_VCODEC_CLK_SEL	(1<<10)
 	if (of_machine_is_compatible("rockchip,rk3126"))
-		writel_relaxed(readl_relaxed(RK_GRF_VIRT + RK312X_GRF_SOC_CON1)
-			| BIT_VCODEC_CLK_SEL | (BIT_VCODEC_CLK_SEL << 16),
-			RK_GRF_VIRT + RK312X_GRF_SOC_CON1);
+		regmap_write(pservice->grf, RK312X_GRF_SOC_CON1,
+			BIT_VCODEC_CLK_SEL | (BIT_VCODEC_CLK_SEL << 16));
 
 #if VCODEC_CLOCK_ENABLE
 	if (pservice->aclk_vcodec)
-- 
2.35.3

